### PR TITLE
Add atomic swaps to Matching Engine README.

### LINKED ISSUE #305: Implement an atomic-swap application
# Motivation

Linera should support DeFi applications, and having some initial examples implementing DeFi functionality can help with the development of the execution layer.

A simple DeFi application is being able to swap two different token types and amounts between two different users.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

We want to demonstrate how to do atomic swaps on Linera.

## Proposal

Extend the Matching Engine README to show how to configure the chain for atomic swaps.

For convenience, and consistency with other commands, make the `--chain-id` argument for `change-application-permissions` optional.

## Test Plan

The README is tested by `test_script_in_readme`. I also tried it manually.

## Links

- Closes #305 
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: examples/matching-engine/README.md [hunk 1]
```diff
@@ -138,13 +146,121 @@ mutation ExecuteOrder {
...
 }
 ```
 
+
+### Atomic Swaps
+
+If you send tokens to a chain owned by someone else, you rely on them for liveness:
+If they don't handle your messages, you don't have access to your tokens.
+
+If the number of parties who want to swap tokens is limited, this can be addressed
```
Comment (ma2bd) [inline examples/matching-engine/README.md:164]: Maybe insert something like: "Fortunately, Linera provides a solution based on temporary chains."

File: examples/matching-engine/README.md [hunk 2]
```diff
@@ -138,13 +146,121 @@ mutation ExecuteOrder {
...
+by making them all chain owners, allowing only matching engine operations on the chain,
+and allowing only the matching engine to close the chain.
+
+```bash
+kill %%    # Kill the service so we can use CLI commands for chain 1.
+
+linera --wait-for-outgoing-messages change-ownership \
```
Comment (ma2bd) [inline examples/matching-engine/README.md:171]: We could set an environment variable `LINERA_WAIT_FOR_OUTGOING_MESSAGES` (in another PR)

**Review by ma2bd — APPROVED** [review]
Nice!

File: examples/matching-engine/README.md (hunk 2 above)
Comment (afck) [reply examples/matching-engine/README.md:173]: That's not much shorter, though. :smile:
We could call it `LINERA`? The outgoing messages option is mostly important for the test anyway, I think?

**Pull request merged**